tests/topotests/ospf6-topo1/test_ospf6_topo1.py

```diff
@@ -372,23 +383,19 @@ def test_linux_ipv6_kernel_routingTable():
 
             assert failures == 0, "Linux Kernel IPv6 Routing Table verification failed for router r%s:\n%s" % (i, diff)
 
-    # For debugging after starting FRR/Quagga daemons, uncomment the next line
-    # CLI(net)
-
 
 def test_shutdown_check_stderr():
-    global fatal_error
-    global net
 
-    # Skip if previous fatal error condition is raised
-    if (fatal_error != ""):
-        pytest.skip(fatal_error)
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip('skipped because of router(s) failure')
 
     if os.environ.get('TOPOTESTS_CHECK_STDERR') is None:
```

Comment:
We don't have a standard function in `topogen` to do this, but maybe it is easy to implement one seeing this pattern.

Check out how the memory leak function was implemented: https://github.com/FRRouting/frr/blob/master/tests/topotests/lib/topogen.py#L750